common_helper_extraction/fs.py

```diff
@@ -15,50 +15,20 @@
     You should have received a copy of the GNU General Public License
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 '''
-import logging
-from struct import calcsize, unpack
+from .jffs import extract_jffs
+from .sqfs import extract_sqfs
+from .ubifs import extract_ubifs
+from .yaffs import extract_yaffs
 
 SQFS_MAGIC_STRINGS = [b'sqsh', b'qshs', b'shsq', b'hsqs']
 SQFS_SIZE_BUFFER_OFFSET = 0x28
 SQFS_SIZE_BUFFER_TYPE = 'Q'
 
 
-def extract_fs(input_data: bytes, magic_strings: list, size_buffer_offset: int, size_buffer_type: str) -> list:
+def extract_fs(input_data: bytes) -> list:
     fs_sections = list()
-    for fs_magic in magic_strings:
-        fs_sections.extend(_get_fs_sections_with_magic(input_data, fs_magic, size_buffer_offset, size_buffer_type))
+    fs_sections.extend(extract_sqfs(input_data))
+    fs_sections.extend(extract_yaffs(input_data))
+    fs_sections.extend(extract_ubifs(input_data))
+    fs_sections.extend(extract_jffs(input_data))
```

Comment:
Changed it to
```Python3
FS_EXTRACTORS = [extract_sqfs, extract_yaffs, extract_ubifs, extract_jffs]
fs_sections = list(filter(None, [extractor(input_data) for extractor in FS_EXTRACTORS]))
```
Additional filter is necessary to avoid empty lists in ```fs_sections```

Reply:
This would be a nice case for the new walrus operator in Python 3.8:
```python
[sections for extractor in FS_EXTRACTORS if (sections := extractor(input_data))]
```
Sadly, we will not be able to use it for years.